specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPacks_API.json

```diff
@@ -0,0 +1,376 @@
+            }
+          }
+        },
+        "x-ms-examples": {
+          "QueryPackUpdateTagsOnly": {
+            "$ref": "./examples/QueryPacksUpdateTagsOnly.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "QueryPacksResource": {
+      "properties": {
+        "id": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Azure resource Id"
+        },
+        "name": {
+          "type": "string",
+          "description": "Azure resource name",
+          "readOnly": true
+        },
+        "type": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Azure resource type"
+        },
+        "location": {
+          "type": "string",
+          "description": "Resource location",
+          "x-ms-mutability": ["create", "read"]
+        },
+        "tags": {
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Resource tags"
+        }
+      },
+      "required": ["location"],
+      "x-ms-azure-resource": true,
+      "description": "An azure resource object"
+    },
+    "TagsResource": {
+      "properties": {
+        "tags": {
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Resource tags"
+        }
+      },
+      "description": "A container holding only the Tags for a resource, allowing the user to update the tags on a QueryPack instance."
+    },
+    "LogAnalyticsQueryPack": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "description": "Properties that define a Log Analytics QueryPack resource.",
+          "$ref": "#/definitions/LogAnalyticsQueryPackProperties"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/QueryPacksResource"
+        }
+      ],
+      "required": ["properties"],
+      "description": "An Log Analytics QueryPack definition."
+    },
+    "LogAnalyticsQueryPackProperties": {
+      "description": "Properties that define a Log Analytics QueryPack resource.",
+      "properties": {
+        "queryPackId": {
+          "type": "string",
+          "description": "The unique ID of your application. This field cannot be changed.",
+          "readOnly": true
+        },
+        "name": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Unique name for your Query Pack."
+        },
+        "timeCreated": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Creation Date for the Log Analytics QueryPack, in ISO 8601 format.",
+          "format": "date-time"
+        },
+        "timeModified": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Last modified date of the Log Analytics QueryPack, in ISO 8601 format.",
+          "format": "date-time"
+        },
+        "tenantId": {
```

Comment:
Why persist this? Do you update it if the subscription is moved? This is always resolvable from the subscriptionId

Reply:
thanks, fixed

Reply:
Kris's comment was about the "tenantId" here.  Is there a reason it needs to be stored with the resource and not resolved from the subscription registration?

Reply:
for now, your'e right, there's no need to explicitly contain it in the resource.